cmd/bacalhau/docker_run.go

```diff
@@ -315,20 +327,30 @@ var dockerRunCmd = &cobra.Command{
 		if err != nil {
 			return fmt.Errorf("error executing job: %s", err)
 		}
-
-		err = ExecuteJob(ctx,
-			cm,
-			cmd,
-			jobSpec,
-			jobDeal,
-			ODR.IsLocal,
-			ODR.WaitForJobToFinish,
-			ODR.DockerRunDownloadFlags)
-
+		if !ODR.DryRun {
+			err = ExecuteJob(ctx,
+				cm,
+				cmd,
+				jobSpec,
+				jobDeal,
+				ODR.IsLocal,
+				ODR.WaitForJobToFinish,
+				ODR.DockerRunDownloadFlags)
+		}
 		if err != nil {
 			return fmt.Errorf("error executing job: %s", err)
 		}
```

Comment:
We need to move this error block inside the `!ODR.DryRun` otherwise this is refering to the error before and so is having no affect